vraptor-core/src/main/java/br/com/caelum/vraptor/observer/ExecuteMethod.java

```diff
@@ -79,6 +85,11 @@ public Void call() throws Exception {
 
 				log.debug("Invoking {}", reflectionMethod);
 				Object instance = event.getControllerInstance();
+				
+				Class<?> declaringClass = reflectionMethod.getDeclaringClass();
+				if (!declaringClass.isAssignableFrom(instance.getClass())) {
+					instance = container.instanceFor(declaringClass);
```

Comment:
We should fix ExecutedMethod for the "forwarded" request, otherwise we'll be hiding similar bugs elsewhere in the code. 

A forward to String does a real container forward that's why we get the bug?